Automattic/wp-calypso
# SDK: Try wp.data with calypso state tree

This creates a `custom-store-plugin` which is a general-purpose plugin which allows any pre-existing redux store to be used for a given wp.data store instead of wp.data creating one itself. This allows for two things:

1. Stores can use enhancers and middleware, which is not possible in the
default wp.data configuration.
2. wp.data stores can share redux stores.

Note: This also creates an `internals-plugin`, which is necessary to avoid the reimplementing of internals within the `calypso-registry-plugin`. The `use` function of the `internals-plugin`
will be submitted as a PR to core Gutenberg as well so perhaps it will not be necessary for long.

**For review purposes, the `internals-plugin.js` implementation is identical to the `@wordpress/data` registry implementation except for the `use` function.**

To Test:
1. Start calypso in local dev mode as usual.
2. Go to `http://calypso.localhost:3000/me/account` and change the "Admin Color Scheme"
3. Behavior should be the same, but it's using `withSelect` now.

Note: CI Failure is because of a reference to `window` in the `@wordpress/data` registry code. I've added a PR to fix this: https://github.com/WordPress/gutenberg/pull/9261
